carbon-design-system/carbon-for-ibm-dotcom
# fix(cloud-masthead): Cloud Masthead + Mega Menu design enhancement follow up fixes

### Related Ticket(s)

DDS Ticket: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6240
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-1972

### Description
- This PR implements design QA feedback that was missing from this PR - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5922

### Design Spec
- Masthead: https://ibm.ent.box.com/file/786859527229
- Mega Menu: https://ibm.ent.box.com/file/774092317795

### Changelog

![Screen Shot 2021-07-13 at 4 14 32 PM](https://user-images.githubusercontent.com/1815714/125518931-faf73b27-7419-4261-aef9-a0470a842505.png)

**Notes**

- I've adjusted the hamburger/mobile menu to be visible within the `959px` breakpoint instead of `800px` as to how we currently have it. This particular change is being applied to both the Cloud and Global version as scoping the styling means that we have to extend several components in order to have several hosts/classes available that can be targeted in Web Components.

**Testing Instructions**
- Visit the `Cloud Masthead` Storybook variant - `/?path=/story/components-cloud-masthead--default`
- Ensure that the design QA feedback outlined in this issue  (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6240) under the "Pre-launch Must-haves" are addressed.
